packages/merge-on-green/src/merge-on-green.ts

```diff
@@ -331,6 +331,98 @@ handler.checkPRMergeability = async function checkPRMergeability(
   }
 };
 
+/**
+ * For a given repository, looks through all the PRs and checks to see if they have a MOG label
+ * @param owner the owner of the repo
+ * @param repo the repo name
+ * @param context the context of the webhook payload
+ * @returns void
+ */
+handler.pickUpPR = async function pickUpPR(
+  owner: string,
+  repo: string,
+  github: Context['github']
+) {
+  const prs = await github.paginate(await github.pulls.list, {
+    owner,
+    repo,
+    state: 'open',
+  });
+
+  for (const pr of prs) {
+    const labels = await github.paginate(
+      await github.issues.listLabelsOnIssue,
+      {
+        owner,
+        repo,
+        issue_number: pr.number,
+      }
+    );
+
+    const labelFound = labels.find(
+      (label: Label) =>
+        label.name === MERGE_ON_GREEN_LABEL ||
+        label.name === MERGE_ON_GREEN_LABEL_SECURE
+    );
+
+    if (labelFound) {
+      // check to see if the owner has branch protection rules
+      let branchProtection: string[] | undefined;
+      try {
+        branchProtection = (
+          await github.repos.getBranchProtection({
+            owner,
+            repo,
+            branch: 'master',
+          })
+        ).data.required_status_checks.contexts;
+        logger.info(
+          `checking branch protection for ${owner}/${repo}: ${branchProtection}`
+        );
+      } catch (err) {
+        err.message = `Error in getting branch protection\n\n${err.message}`;
+        await github.issues.createComment({
+          owner,
+          repo,
+          issue_number: pr.number,
+          body:
+            "Your PR doesn't have any required checks. Please add required checks to your master branch and then re-add the label. Learn more about enabling these checks here: https://help.github.com/en/github/administering-a-repository/enabling-required-status-checks.",
+        });
+        logger.error(err);
```

Comment:
Should this logic share the same code as the webhook handling?

Reply:
👍 I agree, I think the logic could be simplified if we:

1. check to see if the PR is already tracked in datastore (_if it is tracked do nothing._).
2. if the PR is not tracked, but an automerge label exists, make a synthetic call to `app.on('pull_request.labeled', async context`; I think you could do this by looking at the fields you use from context, and then making sure you populate these in a fake context object you pass.

Reply:
I'm not sure about faking an webhook - we could probably extract that common logic into a function to execute from both paths.

Reply:
I simplified the logic here!